test/library/standard/CommDiagnostics/commDiagsTable.na-none.good

```diff
@@ -33,38 +33,38 @@
 |      2 | 10000 |           10000 |             0 |
 |      3 | 10000 |           10000 |             0 |
 
-| locale | get | get_nb | put | put_nb | test_nb | wait_nb | try_nb |      amo | execute_on | execute_on_fast | execute_on_nb |
-| -----: | --: | -----: | --: | -----: | ------: | ------: | -----: | -------: | ---------: | --------------: | ------------: |
-|      0 |   0 |      0 |   0 |      0 |       0 |       0 |      0 | unstable |          0 |               0 |             3 |
-|      1 |   0 |      0 |   1 |      0 |       0 |       0 |      0 | unstable |          0 |               1 |             0 |
-|      2 |   0 |      0 |   1 |      0 |       0 |       0 |      0 | unstable |          0 |               1 |             0 |
-|      3 |   0 |      0 |   1 |      0 |       0 |       0 |      0 | unstable |          0 |               1 |             0 |
+| locale | get | get_nb | put | put_nb | test_nb | wait_nb | try_nb |      amo | execute_on | execute_on_fast | execute_on_nb | cache_get_hits | cache_get_misses | cache_put_hits | cache_put_misses |
+| -----: | --: | -----: | --: | -----: | ------: | ------: | -----: | -------: | ---------: | --------------: | ------------: | -------------: | ---------------: | -------------: | ---------------: |
+|      0 |   0 |      0 |   0 |      0 |       0 |       0 |      0 | unstable |          0 |               0 |             3 |              0 |                0 |              0 |                0 |
+|      1 |   0 |      0 |   1 |      0 |       0 |       0 |      0 | unstable |          0 |               1 |             0 |              0 |                0 |              0 |                0 |
+|      2 |   0 |      0 |   1 |      0 |       0 |       0 |      0 | unstable |          0 |               1 |             0 |              0 |                0 |              0 |                0 |
+|      3 |   0 |      0 |   1 |      0 |       0 |       0 |      0 | unstable |          0 |               1 |             0 |              0 |                0 |              0 |                0 |
```

Comment:
A couple of questions:

* Once `--cache-remote` is on by default, will any of the comm diags table tests print out non-zero values for cache stats?  Will such numbers be stable / deterministic, or unstable (like amo's)?
* Would it be more natural/useful to interleave the cache_get/put stats with the vanilla get/put stats? (vs. having them off in their own section of the table like this?)  Or should we reorder more significantly such that the execute_on stuff comes first, then the puts/gets (w/ cached stuff interleaved or just following the current categories)?

Reply:
> Once --cache-remote is on by default, will any of the comm diags table tests print out non-zero values for cache stats?

generally speaking, yes

> Will such numbers be stable / deterministic, or unstable (like amo's)?

They will be more stable than AMOs but probably not completely deterministic. They might vary by a few / small percentage. One cause of minor differences is that the cache reads a "cache line" at a time and so adjacent accesses might be hits or misses depending on the starting address. There is a similar problem for cache pages when the cache does readahead. (Cache pages are 1KiB and cache lines are 64B).

(All that said I could try to gather more information about this in practice).

> Would it be more natural/useful to interleave the cache_get/put stats with the vanilla get/put stats? (vs. having them off in their own section of the table like this?) Or should we reorder more significantly such that the execute_on stuff comes first, then the puts/gets (w/ cached stuff interleaved or just following the current categories)?

It might be obvious but in this PR I wasn't really focused on how these tables look and more or less went with the changes that seemed obvious to me. I don't have any real preference about the order but I could argue for having them in their own section or for having get hits/misses next to gets. I think both have advantages.

Reply:
I've noted in the PR message that reordering these fields/columns is future work.